tests/testlib/s2n_connection_test_utils.c

```diff
@@ -180,17 +180,23 @@ int s2n_connections_set_io_pair(struct s2n_connection *client, struct s2n_connec
 
 int s2n_io_pair_close(struct s2n_test_io_pair *io_pair)
 {
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
-    POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));
+    if (io_pair->client != CLOSED_FD) {
+        POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_CLIENT));
+    }
+    if (io_pair->server != CLOSED_FD) {
+        POSIX_GUARD(s2n_io_pair_close_one_end(io_pair, S2N_SERVER));
+    }
```

Comment:
Stealing Sam's previous comment: https://github.com/aws/s2n-tls/pull/4833#discussion_r1796098930
>  I also wonder if the >-1 validity check should go in s2n_io_pair_close_one_end() rather than s2n_io_pair_close(), in case s2n_io_pair_close_one_end() is called twice as well.

Reply:
The change is moved to [PR#4841](https://github.com/aws/s2n-tls/pull/4841).